tests/static-check/run.sh

```diff
@@ -19,15 +21,37 @@ function create_image() {
   buildah run $c -- dnf -q -y install "@C Development Tools and Libraries" clang cppcheck which diffutils file >/dev/null 2>&1
   buildah run $c -- dnf -q -y install pcre-devel pcre2-devel openssl-devel libxml2-devel pam-devel lmdb-devel libacl-devel libyaml-devel curl-devel libvirt-devel >/dev/null 2>&1
   buildah run $c -- dnf clean all >/dev/null 2>&1
+
+  # Copy checksum of this file into container
+  sha256sum $0 > $SUM_FILE
+  buildah copy $c $SUM_FILE >/dev/null
+  rm $SUM_FILE
+
   buildah commit $c cfengine-static-checker-f$STATIC_CHECKS_FEDORA_VERSION >/dev/null 2>&1
   echo $c
 }
 
 set -x
 
-# TODO: check how old the image is and recreate if it's too old
 if buildah inspect cfengine-static-checker-f$STATIC_CHECKS_FEDORA_VERSION >/dev/null 2>&1; then
   c=$(buildah from cfengine-static-checker-f$STATIC_CHECKS_FEDORA_VERSION)
+
+  # Recreate the image if the checksum of this file has changed 
+  if [[ `buildah run $c ls $SUM_FILE` == $SUM_FILE ]]; then
+    SUM_A=$(sha256sum $0)
+    SUM_B=$(buildah run $c cat $SUM_FILE)
+    if [[ $SUM_A != $SUM_B ]]; then
+      echo "Recreating image due to mismatching checksum..."
+      IMAGE_ID=$(buildah inspect $c | jq -r '.FromImageID')
+      buildah rmi --force $IMAGE_ID >/dev/null
+      c=$(create_image)
```

Comment:
Something should `stop` and `rm` the `$c` container, right? Ideally before removing its image.

Reply:
So the `--force` "option will cause Buildah to remove all containers that are using the image before removing the image from the system" (see [buildah-rmi.1.md](https://github.com/containers/buildah/blob/main/docs/buildah-rmi.1.md)). Also, I'm not familiar with any command to stop containers Buildah.

Reply:
oh, cool! I thought it would leave the container running and just force-remove the image. Perhaps we could add a comment? Thanks!

Reply:
Yup, there should absolutely be a comment explaining this. See https://github.com/cfengine/core/pull/5653